internal/controllers/topology/cluster/desired_state.go

```diff
@@ -718,20 +737,15 @@ type templateToInput struct {
 // of adding required labels (cluster, topology), annotations (clonedFrom)
 // and assigning a meaningful name (or reusing current reference name).
 func templateToObject(in templateToInput) (*unstructured.Unstructured, error) {
-	// NOTE: The cluster label is added at creation time so this object could be read by the ClusterTopology
-	// controller immediately after creation, even before other controllers are going to add the label (if missing).
-	labels := map[string]string{}
-	labels[clusterv1.ClusterLabelName] = in.cluster.Name
-	labels[clusterv1.ClusterTopologyOwnedLabel] = ""
-
```

Comment:
How about keeping this and doing a `mergeMap` with in.Labels and using that as in the `GenerateTemplate` function. This way we wont have to duplicate this everywhere templateToObjects is used?

Reply:
> @MaxFedotov the proposal is now merged. [Proposal here.](https://github.com/kubernetes-sigs/cluster-api/blob/main/docs/proposals/20221003-In-place-propagation-of-Kubernetes-objects-only-changes.md#metadata-propagation) The proposal contains all the details for the questions on this PR.
> 
> Do you have bandwidth to continue working on this? I am happy to pick up the pending work here, in case you do not have bandwidth 🙂

Hi @ykakarap! I would be grateful if you take this on. Unfortunately i don't have free time right now to continue with it :(